src/main/java/org/icgc/argo/workflow_management/service/NextflowService.java

```diff
@@ -14,42 +24,43 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.UUID;
-
-import static java.util.Objects.nonNull;
-import static org.icgc.argo.workflow_management.util.ParamsFile.createParamsFile;
-import static org.icgc.argo.workflow_management.util.Reflections.createWithReflection;
-import static org.icgc.argo.workflow_management.util.Reflections.invokeDeclaredMethod;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
 
 @Slf4j
 @Service(value = "nextflow")
 public class NextflowService implements WorkflowExecutionService {
 
   @Autowired private NextflowProperties config;
 
+  private Scheduler scheduler = Schedulers.newElastic("nextflow-service");
+
   public Mono<RunsResponse> run(WESRunParams params) {
-    // TODO: This is not really fluxing, make it flux
-    return Mono.create(
-        callback -> {
-          try {
-            val cmd = createCmd(createLauncher(), params);
-            val driver = createDriver(cmd);
-
-            // Run it!
-            driver.run(params.getWorkflowUrl(), Arrays.asList());
-            val exitStatus = driver.shutdown();
-            val response = new RunsResponse(exitStatus == 0 ? cmd.getRunName() : "Error!");
-
-            callback.success(response);
-          } catch (Exception e) {
-            callback.error(e);
-          }
-        });
+    return Mono.fromSupplier(
+            () -> {
+              try {
+                return this.startRun(params);
+              } catch (Exception e) {
+                log.error("startRun error", e);
+                throw new RuntimeException(e.getMessage());
+              }
+            })
+        .map(RunsResponse::new)
+        .subscribeOn(scheduler);
+  }
+
+  private String startRun(WESRunParams params) throws Exception {
```

Comment:
lets add some semantics to the exception by using something with a more specific exception type, even if you have to define your own class that extends `Exception`. I do like the use of a checked exception though.